swift/ql/src/queries/Security/CWE-321/HardcodedEncryptionKey.ql

```diff
@@ -0,0 +1,70 @@
+/**
+ * @name Hard-coded encryption key
+ * @description Using hardcoded keys for encryption is not secure, because potential attacker can easiy guess them.
+ * @kind path-problem
+ * @problem.severity error
+ * @security-severity 8.1
+ * @precision high
+ * @id swift/hardcoded-key
+ * @tags security
+ *       external/cwe/cwe-321
+ */
+
+import swift
+import codeql.swift.dataflow.DataFlow
+import DataFlow::PathGraph
+
+/**
+ * An `Expr` that is used to initialize a key.
+ */
+abstract class KeySource extends Expr { }
+
+/**
+ * The creation of a literal byte array.
+ */
+class ByteArrayLiteralSource extends KeySource {
+  ByteArrayLiteralSource() { this = any(ArrayExpr arr | arr.getType().toString() = "Array<UInt8>") }
```

Comment:
I _think_ `toString` does the same as `getName` here, right? In that case, I'd prefer using `getName` here (since we like to be able to mess with the output of `toString` without fear of breaking query logic).

Reply:
Agreed, and it does behave the same way. Confirmed by re-running the tests, so I've changed it to `getName` instead.